go.mod

```diff
@@ -57,6 +57,7 @@ require (
 	github.com/go-ego/gse v0.80.2
 	github.com/googleapis/gax-go/v2 v2.12.2
 	github.com/hashicorp/go-hclog v1.6.2
+	github.com/hashicorp/golang-lru/v2 v2.0.7
```

Comment:
Not sure if this is better, but I've came across this library when thinking about BM25 and cache
https://github.com/maypok86/otter